added docu for second prometheus instance (#9837)
kyma-project/kyma: docs/monitoring/03-02-istio.md

---
title: Istio Monitoring
type: Details
---

> The monitoring chart is pre-configured to collect all metrics relevant for observing the in-cluster [Istio](https://istio.io/latest/docs/concepts/observability/) service mesh, including the proxy-level, service-level and control-plane metrics.

Reviewer: ```suggestion
The monitoring chart is pre-configured to collect all metrics relevant for observing the in-cluster [Istio](https://istio.io/latest/docs/concepts/observability/) Service Mesh, including the proxy-level, service-level, and control-plane metrics.
```

> The concept for collecting the [service-level](https://istio.io/latest/docs/concepts/observability/#service-level-metrics) metrics is build around the Istio Proxy (implemented by Envoy), which is counting decentralized all communication inside the service mesh. After scraping these high cardinality metrics from the envoys, still an aggregation on service level is required to get the final service related metrics.

Reviewer: ```suggestion
The concept of collecting the [service-level](https://istio.io/latest/docs/concepts/observability/#service-level-metrics) metrics is built around the Istio Proxy implemented by Envoy. Istio Proxy collects all decentralized communication details inside the service mesh. After scraping these high cardinality metrics from the envoys, the metrics need to be additionally aggregated on a service level to get the final service-related details.
```

Author: Istio Proxy collects decentralized all communication details

Author: I adjusted your comment to easily merge it, please review

> Following the [Istio Observability Best Practice](https://istio.io/latest/docs/ops/best-practices/observability/) the scraping and aggregating of the service-level metrics is done in a dedicated prometheus instance. That instance has the smallest possible data retention time configured as the raw metrics being scraped from the Istio Proxies have a high cardinality and are not further required to be kept. Instead, the aggregated metrics will be then scraped by the main prometheus instance via the federate endpoint.

Reviewer: ```suggestion
Following the [Istio's observability best practice](https://istio.io/latest/docs/ops/best-practices/observability/), the scraping and aggregation of the service-level metrics is done in a dedicated Prometheus instance. That instance has the smallest possible data retention time configured as the raw metrics scraped from the Istio Proxies have high cardinality and are not further required to be kept. Instead, the main Prometheus instance scapes the aggregated metrics through the `/federate` endpoint.
```

> The istop-related instance is a Deployment with name 'monitoring-prometheus-istio-server'. It has a small data retention time which should not be changed, no PersistentVolume attached and it has a hardcoded configuration. Resources like a ServiceMonitor will never be picked up by that instance.

Reviewer: ```suggestion
The Istio-related instance is a Deployment named `monitoring-prometheus-istio-server`. This instance has a short data retention time and hardcoded configuration that should not be changed. It also has no PersistentVolume attached. This instance never collects metrics from such resources as ServiceMonitors.
```

Reviewer: if that is what you meant by "picking up"? I think it would also be nice to explain what you mean by `resources like` - what is that category of resources?

> ![Istio Monitoring](./assets/monitoring-istio.svg)

Reviewer: As for that diagram, I think that there should be a sentence explaining what this diagram is to present. For example, `See the diagram for a broader view of how the Istio-related instance fits into the whole monitoring flow in Kyma:` or sth like that,

As for the diagram itself:
- One arrow should be oriented towards Grafana 
- Two `scrape metrics` have an additional background that should be removed
- Pls capitalize `grafana` to be consistent throughout the diagram
- Pls change `discovered via ServiceMonitor resources` to `discovered through ServiceMonitor resources` (we avoid `via` in docs, as per the guidelines)
- Pls change `having envoy port exposed` into  `with the exposed Envoy port` or `having the Envoy port exposed`
![image](https://user-images.githubusercontent.com/40655785/97880719-cddef780-1d21-11eb-830a-6d2db27fd338.png)

Author: Everything applied besides firts bullet point. Indeed grafana is calling prometheus
